### ISSUE TITLE
Unhandled error when fare checking and fare class is sold out

### ISSUE THREAD

**Issue by pcarn** [jdholtz/auto-southwest-check-in#146]
### Version

5.1

### Description

When fare checking, and the purchased fare class is sold out, there is a KeyError that can be handled better.

### To Reproduce

Book a ticket with Wanna Get Away fare
Wait for Wanna Get Away fare to be sold out, but other fare classes to be available

### Expected Behavior

Instead of `Unexpected error`, I think it can look at the cheapest for the other fare classes, or simply print a message that it is sold out.

### Relevant logs and program output

```sh
2023-09-07 17:23:37 ERROR Process-1[reservation_monitor:106]: Unexpected error during fare check: KeyError('No fare type found matching WGA')
Traceback (most recent call last):
  File "/app/lib/reservation_monitor.py", line 100, in _check_flight_fares
    fare_checker.check_flight_price(flight)
  File "/app/lib/fare_checker.py", line 31, in check_flight_price
    flight_price = self._get_flight_price(flight)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/fare_checker.py", line 58, in _get_flight_price
    return self._get_matching_fare(new_flight["fares"], fare_type)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/fare_checker.py", line 159, in _get_matching_fare
    raise KeyError(f"No fare type found matching {fare_type}")
KeyError: 'No fare type found matching WGA'
```

**Comment by jdholtz** [comment]
This should be fixed in the `develop` branch. In this case, since the fare is not available, the price difference will be set to 0 and will print a log message showing that as well.
